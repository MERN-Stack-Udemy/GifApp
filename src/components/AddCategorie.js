import React, { useState } from "react";
import PropTypes from 'prop-types'


export const AddCategorie = ( {setCategories} ) => {

  const [inputvalue, setInputvalue] = useState(''); // ''

  const handleChange = (e) => {
    setInputvalue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if(inputvalue.trim().length > 2){
      setCategories(categories => [...categories, inputvalue])
      setInputvalue( "" );
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input 
        type = "text" 
        value = {inputvalue} 
        onChange = {handleChange} />
    </form>
  );

};

AddCategorie.propTypes = {
  setCategories : PropTypes.func.isRequired,
}
