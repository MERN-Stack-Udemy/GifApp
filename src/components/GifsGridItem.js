import React from 'react'

export const GifsGridItem = ( {image, title} ) => {
  
  return (
    <div className="card">
      <img 
        src={image.url}
        alt= {title}
      />
      <p>{title}</p>
    </div>
  )
}
