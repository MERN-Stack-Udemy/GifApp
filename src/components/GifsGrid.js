import React, { useEffect, useState } from 'react'
import { getGifs } from '../Helpers/getGifs';
import { GifsGridItem } from './GifsGridItem';

export const GifsGrid = ({ categorie }) => {

  const [images, setImages] = useState([]);

  useEffect(() => {

    getGifs(categorie)
    .then( setImages )

  }, [categorie] );

 


  return (
    <>
      <h3>{categorie}</h3>
      <div className="card-grid">
        {images.map( (img) => (
          <GifsGridItem 
            key = {img.id} 
            {...img}
          />
        ))}
        
      </div>
    </>
  )
}
