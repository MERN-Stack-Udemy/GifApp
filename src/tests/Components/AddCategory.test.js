import { shallow } from "enzyme"
import { AddCategorie } from "../../components/AddCategorie"

describe('Test on <AddCategory /> component', () => {

  const setCategories = () => {};
  const wrapper = shallow(<AddCategorie setCategories = { setCategories} />)
  
  test('should test component to match snapshot', () => {
    expect( wrapper ).toMatchSnapshot();
  })
  
  test('should change input box value', () => {
    const input = wrapper.find('input');
    const value = 'hello world';

    input.simulate('change', { target:{ value }} );
    expect( wrapper.find('p').text() ).toBe( value );
  })
  


})
