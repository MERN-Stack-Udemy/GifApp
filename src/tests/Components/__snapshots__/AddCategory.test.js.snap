// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AddCategory /> component should test component to match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
